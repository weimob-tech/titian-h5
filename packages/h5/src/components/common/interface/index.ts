export type OmitType<T, P> = Pick<T, { [key in keyof T]: T[key] extends P ? never : key }[keyof T]>;

export type Key = string | number | symbol;

export type IAnyObject = { [key: Key]: unknown };

export type IAnyMap = Map<Key, unknown>;

export type DistanceUnit = 'px' | 'em' | 'rem' | 'vw' | 'vh' | '%';

export type IPosition = {
  bottom?: `${number}${DistanceUnit}`;
  top?: `${number}${DistanceUnit}`;
  left?: `${number}${DistanceUnit}`;
  right?: `${number}${DistanceUnit}`;
};
