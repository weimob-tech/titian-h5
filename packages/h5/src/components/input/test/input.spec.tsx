import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiInput } from '../index';
import { TestContainer } from '../../common/test/container';
import { sleep } from '../../common/test';
describe('渲染 ti-input', () => {
  it('填充模式', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input label="左侧标题" placeholder="用户输入中文案" />
          <ti-input disabled label="禁用" />
          <ti-input readOnly label="只读" />
          <ti-input maxlength={20} label="最大输入长度20" />
          <ti-input divider={false} label="不显示底部分割线" />
          <ti-input required label="必填" />
          <ti-input clearable={false} label="不显示清除按钮" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('输入类型', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input type="text" label="文本" />
          <ti-input type="number" label="数字" />
          <ti-input type="safe-password" label="密码安全输入键盘" />
          <ti-input type="digit" label="带小数点的数字键盘" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('输入对齐方式', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input text-align="left" label="标题" />
          <ti-input text-align="right" label="标题" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('label左侧搭配图标', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input prefix-icon="home" label="标题" />,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('输入框左右使用插槽', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input label="手机号">
            <div slot="prefix">+ 86</div>
          </ti-input>
          <ti-input label="验证码">
            <div slot="suffix">发验证码</div>
          </ti-input>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('错误提示', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input error value="输错了" label="标题" />
          <ti-input error-message="手机号格式错误" value="123213" label="手机号" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('键盘确认按钮文字', async () => {
    const page = await newSpecPage({
      components: [TiInput, TestContainer],
      template: () => (
        <test-container>
          <ti-input confirm-type="done" label="完成" />
          <ti-input confirm-type="send" label="发送" />
          <ti-input confirm-type="search" label="搜索" />
          <ti-input confirm-type="next" label="下一项" />
          <ti-input confirm-type="go" label="前往" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('点击清除按钮', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="text" label="文本" value="111111" />,
    });
    const input = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    input.focus();
    await page.waitForChanges();
    const dom = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-input-icon-box');
    dom?.dispatchEvent(new Event('touchStart', {}));
    await sleep(1000);
    dom?.dispatchEvent(new Event('touchEnd', {}));
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('禁用', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="23" label="数字" disabled />,
    });
    const input = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    input.focus();
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('自动聚焦', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="23" label="数字" autofocus />,
    });
    const input = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    input.focus();
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('格式化1', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="23" label="数字" />,
    });
    const input = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    input.focus();
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
  it('格式化2', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="0123" label="数字" />,
    });
    const input = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    input.focus();
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('输入', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="0123" label="数字" />,
    });
    const dom = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    dom?.dispatchEvent(new CustomEvent('input', { detail: '12' }));
    await sleep(1000);
    dom?.dispatchEvent(new Event('blur'));
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('键盘确认', async () => {
    const page = await newSpecPage({
      components: [TiInput],
      template: () => <ti-input type="number" value="0123" label="数字" />,
    });
    const dom = page.root.shadowRoot.querySelector<HTMLInputElement>('.titian-input-input');
    dom?.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
    await sleep(200);
    dom?.dispatchEvent(new KeyboardEvent('keyup', { key: 'Enter' }));
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
});
