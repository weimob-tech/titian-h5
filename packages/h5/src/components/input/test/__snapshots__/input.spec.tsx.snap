// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-input label左侧搭配图标 1`] = `
<ti-input prefix-icon="home">
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <ti-icon color="#9e9e9e" ext-class="titian-input-left-icon" name="home" size="28"></ti-icon>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            标题
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 填充模式 1`] = `
<test-container>
  <!---->
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              左侧标题
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" placeholder="用户输入中文案">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input titian-input-disabled" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              禁用
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" disabled="" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input titian-input-readOnly" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              只读
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" readonly="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              最大输入长度20
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="20" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input titian-input-borderless" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              不显示底部分割线
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              必填
            </div>
          </div>
          <div class="titian-input-required-icon">
            <ti-icon ext-class="titian-input-icon" name="required" size="24"></ti-icon>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              不显示清除按钮
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
</test-container>
`;

exports[`渲染 ti-input 格式化1 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text" value="23">
      </div>
      <div aria-hidden="true" class="titian-input-icon-box">
        <ti-icon color="#9e9e9e" ext-class="titian-input-icon" name="clear" size="36"></ti-icon>
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 格式化2 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text" value="0123">
      </div>
      <div aria-hidden="true" class="titian-input-icon-box">
        <ti-icon color="#9e9e9e" ext-class="titian-input-icon" name="clear" size="36"></ti-icon>
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 点击清除按钮 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            文本
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" type="text" value="">
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 禁用 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input titian-input-disabled" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" disabled="" inputmode="numeric" maxlength="140" part="" type="text" value="23">
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 自动聚焦 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input autofocus="" class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text" value="23">
      </div>
      <div aria-hidden="true" class="titian-input-icon-box">
        <ti-icon color="#9e9e9e" ext-class="titian-input-icon" name="clear" size="36"></ti-icon>
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 输入 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text" value="0123">
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 输入对齐方式 1`] = `
<test-container>
  <!---->
  <ti-input text-align="left">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              标题
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-left titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input text-align="right">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              标题
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-right titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
</test-container>
`;

exports[`渲染 ti-input 输入框左右使用插槽 1`] = `
<test-container>
  <!---->
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              手机号
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="prefix">
      + 86
    </div>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              验证码
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="suffix">
      发验证码
    </div>
  </ti-input>
</test-container>
`;

exports[`渲染 ti-input 输入类型 1`] = `
<test-container>
  <!---->
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              文本
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" type="text">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              数字
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              密码安全输入键盘
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" type="password">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              带小数点的数字键盘
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="decimal" maxlength="140" part="" type="text">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
</test-container>
`;

exports[`渲染 ti-input 错误提示 1`] = `
<test-container>
  <!---->
  <ti-input>
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              标题
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input titian-input-input-error" inputmode="text" maxlength="140" part="" value="输错了">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input error-message="手机号格式错误">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              手机号
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" inputmode="text" maxlength="140" part="" value="123213">
          <div class="input-text-align-undefined titian-input-error-message">
            <ti-icon ext-class="titian-input-warning-icon" name="warning" size="24"></ti-icon>
            <text>
              手机号格式错误
            </text>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
</test-container>
`;

exports[`渲染 ti-input 键盘确认 1`] = `
<ti-input>
  <mock:shadow-root>
    <div class="titian-input" part="">
      <div class="titian-input-label-container">
        <slot name="prefix-icon"></slot>
        <div class="titian-input-title">
          <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
            数字
          </div>
        </div>
      </div>
      <div class="titian-input-slot">
        <slot name="prefix"></slot>
      </div>
      <div class="titian-input-container">
        <input class="input-text-align-undefined titian-input-input" inputmode="numeric" maxlength="140" part="" type="text" value="0123">
      </div>
      <div class="titian-input-slot">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-input>
`;

exports[`渲染 ti-input 键盘确认按钮文字 1`] = `
<test-container>
  <!---->
  <ti-input confirm-type="done">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              完成
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" enterkeyhint="done" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input confirm-type="send">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              发送
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" enterkeyhint="send" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input confirm-type="search">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              搜索
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" enterkeyhint="search" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input confirm-type="next">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              下一项
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" enterkeyhint="next" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
  <ti-input confirm-type="go">
    <mock:shadow-root>
      <div class="titian-input" part="">
        <div class="titian-input-label-container">
          <slot name="prefix-icon"></slot>
          <div class="titian-input-title">
            <div class="titian-input-title-text" part="" style="--input-line-clamp: 0;">
              前往
            </div>
          </div>
        </div>
        <div class="titian-input-slot">
          <slot name="prefix"></slot>
        </div>
        <div class="titian-input-container">
          <input class="input-text-align-undefined titian-input-input" enterkeyhint="go" inputmode="text" maxlength="140" part="">
        </div>
        <div class="titian-input-slot">
          <slot name="suffix"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-input>
</test-container>
`;
